tests/senml_phydat/main.c

```diff
@@ -0,0 +1,290 @@
+        .senml = senml_s(0)
+    },
+    {
+        .phydat = { { 28, 4, 2021 }, UNIT_DATE, 0 },
+        .senml = senml_s(1619568000),
+    },
+    {
+        .phydat = { { 29, 2, 2020 }, UNIT_DATE, 0 },
+        .senml = senml_s(1582934400),
+    },
+    {
+        .phydat = { { 1, 3, 2020 }, UNIT_DATE, 0 },
+        .senml = senml_s(1583020800),
+    },
+    {
+        .phydat = { { 1, 1, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2208988800),
+    },
+    {
+        .phydat = { { 1, 2, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2206310400),
+    },
+    {
+        .phydat = { { 1, 3, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2203891200),
+    },
+    {
+        .phydat = { { 1, 4, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2201212800),
+    },
+    {
+        .phydat = { { 1, 5, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2198620800),
+    },
+    {
+        .phydat = { { 1, 6, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2195942400),
+    },
+    {
+        .phydat = { { 1, 7, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2193350400),
+    },
+    {
+        .phydat = { { 1, 8, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2190672000),
+    },
+    {
+        .phydat = { { 1, 9, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2187993600),
+    },
+    {
+        .phydat = { { 1, 10, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2185401600),
+    },
+    {
+        .phydat = { { 1, 11, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2182723200),
+    },
+    {
+        .phydat = { { 1, 12, 1900 }, UNIT_DATE, 0 },
+        .senml = senml_s(-2180131200),
+    },
+};
+
+static time_test_t time_tests[] = {
+    {
+        .phydat = { { 0, 0, 0 }, UNIT_TIME, 0 },
+        .senml = senml_s(0)
+    },
+    {
+        .phydat = { { 0, 0, 12 }, UNIT_TIME, 0 },
+        .senml = senml_s(43200)
+    },
+    {
+        .phydat = { { 59, 59, 23 }, UNIT_TIME, 0 },
+        .senml = senml_s(86399)
+    },
+};
+
+static value_test_t value_tests[] = {
+    {
+        .phydat = { { 360, 0, 0 }, UNIT_M, 6 },
+        .senml1 = senml_f(360e6, SENML_UNIT_METER),
+        .senml2 = senml_df(360, 6, SENML_UNIT_METER),
+        .dim = 0
+    },
+    {
+        .phydat = { { 864, 0, 0 }, UNIT_TIME, 2 },
+        .senml1 = senml_f(86400, SENML_UNIT_SECOND),
+        .senml2 = senml_df(864, 2, SENML_UNIT_SECOND),
+        .dim = 0
+    },
+    {
+        .phydat = { { 0, 144, 0 }, UNIT_TIME, 1 },
+        .senml1 = senml_f(1440, SENML_UNIT_MINUTE),
+        .senml2 = senml_df(144, 1, SENML_UNIT_MINUTE),
+        .dim = 1
+    },
+    {
+        .phydat = { { 0, 0, 24 }, UNIT_TIME, 0 },
+        .senml1 = senml_f(24, SENML_UNIT_HOUR),
+        .senml2 = senml_df(24, 0, SENML_UNIT_HOUR),
+        .dim = 2
+    },
+    {
+        .phydat = { { 0, 0, 0 }, UNIT_TEMP_F, 3 },
+        .senml1 = senml_f(255.37, SENML_UNIT_KELVIN),
+        .senml2 = senml_df(0, 3, SENML_UNIT_NONE),
```

Comment:
Quickly skimming over this, shouldn't these three match?
```suggestion
        .phydat = { { 25537, 0, 0 }, UNIT_TEMP_F, -2 },
        .senml1 = senml_f(255.37, SENML_UNIT_KELVIN),
        .senml2 = senml_df(25537, -2, SENML_UNIT_KELVIN),
```

Reply:
Hmm `phydat` is in fahrenheit, `senml1` is in kelvin, so 0F is indeed 255.37K. For the `senml2` I think you are correct though since it would be converted to Kelvin, @silkeh is that correct?

Reply:
I stand corrected, `senml2` is using `phydat_to_senml_decimal` which does not have special handling for `F` so it ends up calling ` phydat_unit_to_senml_unit` with no conversion.

Reply:
Makes sense, sorry for the noise